importer/src/path_canonicalizer.rs

```diff
@@ -646,6 +696,47 @@ fn check_expressions(
         .try_for_each(|e| check_expression(location, e, state, local_variables))
 }
 
+/// Checks paths in a pattern and returns the newly declared variables.
+/// Does not turn "wrongly parsed" enums into variables, that will be done
+/// in canonicalize_inside_pattern.
+fn check_pattern<'b>(
```

Comment:
Would it make sense to call these "check_*" functions "resolve_*"?
What they do is resolve some relative path to an absolute path right?

Reply:
Yeah, it's a bit cryptic because of the architecture of that file.
There is an issue to completely re-do this file, so I would like to keep it like that until the rewrite.

Essentially what the `check_` functions do is they build up a mapping from local names to absolute paths. The mapping is then applied later in the canonicalize_ functions. The re-write is about applying the mapping while it is being created already.